### DISCUSSION TITLE
Developing tessellation from Cellular Automaton Results

### DISCUSSION THREAD

**Discussion by lreigbua** [neperfepx/neper#484] [Q&A] [answered]
Hello,

Do you think it could be possible to create a Neper tessellation using the results from a Cellular Automaton (CA) algorithm?

These are the results we are getting from CA:

![celular automaton](https://user-images.githubusercontent.com/93150422/170522140-8dcf9621-8bb5-416f-b394-396623f985ab.png)

Is there any way to obtain a simplistic representation of these results in Neper by importing information like centroid for the seeds and the grain size?

We would like this in order to  introduce multiscale tessellations and to obtain a conformal mesh.

Thanks for all the help!

**Comment by rquey** [comment, marked as answer]
You can write your volume as a tesr, and then approximate it by a tess using [-morpho "tesr:file(<file_name>)"](https://neper.info/doc/neper_t.html#cmdoption-morpho). This will generate the best convex-cell approximation of your polycrystal. The method uses the grain centroids and sizes for the initial solution and then goes into optimization to minimize the distance between the boundary networks; see https://neper.info/papers.html#id2.

However, in this specific case, given the tortuous grain shapes, I suspect the result will be quite different from the actual polycrystal.
